src/plugins/identity/contractIdentities.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+import * as Weights from "../../core/weights";
+import {type WeightedGraph as WeightedGraphT} from "../../core/weightedGraph";
+import {type NodeContraction, NodeAddress} from "../../core/graph";
+import {nodeContractions} from "./nodeContractions";
+import {type Identity} from "./identity";
+
+/**
+ * Applies nodeContractions to a WeightedGraph.
+ *
+ * This functionality is defined as part of the identity plugin rather than as
+ * a feature of WeightedGraph because it doesn't attempt to contract weights
+ * (it's not clear how to do this in a general principled way). Within the
+ * identity plugin, we do not expect user identities to have weights
+ * associated, so we can contract the graph without attempting to contract the
+ * weights.
```

Comment:
Side-note: speaking of unexpected things. Maybe we should document a plugin developers warning that creating user-to-user edges between plugins could create self-loops when contracting identities?

Reply:
Heh, yeah, add it to the list of stuff we should document whenever we document how to add a new plugin ;)